### PR TITLE
Updated & Documented the video_matching_eval class

### PR DESCRIPTION
**Pull request by DaviidK**
Updated the video_matching_eval class to contain the code used in cascade classifier evaluation. Also added documentation for all methods

### CODE REVIEW

File: Tools/video_matching_eval.cpp [hunk 1]
```diff
@@ -42,9 +42,14 @@ using std::chrono::system_clock;
...
-static const bool USE_META_FILE = false;
+static const bool USE_META_FILE = true;
+
+// Template matching specific parameters
+static const int NUM_MATCHING_METHODS = 8;
+static const int MATCH_METHOD = 2;
+static const bool USE_MASK = false;
 
 // Template matching specific parameters
 static const int NUM_MATCHING_METHODS = 8;
```
Comment (sanasc) [inline Tools/video_matching_eval.cpp]: It looks like this is duplicated above on line 50.

File: Tools/video_matching_eval.cpp [hunk 2]
```diff
@@ -42,9 +42,14 @@ using std::chrono::system_clock;
...
 static const string VIDEO_FILE_PREFIX = "Detection_Algorithm/Data/Video/";
-static const string DETECTION_TYPES[] = { "Template-Matching", "Cascade-Classifier", "Edge-Matching" };
+static const string DETECTION_TYPES[] = { "Template-Matching", "Cascade-Classifier" };
 static const int DETECTION_METHOD = 0;
-static const bool USE_META_FILE = false;
+static const bool USE_META_FILE = true;
+
+// Template matching specific parameters
+static const int NUM_MATCHING_METHODS = 8;
+static const int MATCH_METHOD = 2;
```
Comment (sanasc) [inline Tools/video_matching_eval.cpp]: If you could change this to 3 and true, just to fit the ideal configurations

File: Tools/video_matching_eval.cpp [hunk 3]
```diff
@@ -79,6 +92,9 @@ string getDateTime();
  * - Waits for minimum amount of time and displays the frame
```
Comment (sanasc) [inline Tools/video_matching_eval.cpp]: Sorry I think I left this in when I used the video profiler comments, do you mind changing this to Detection Evaluation or something similar?

**Review by sanasc — APPROVED** [review]
Looks good to me! Thanks for updating my comments as well

**Review by mmunson2 — APPROVED** [review]
Looks great!

**Pull request merged**